third_party/terraform/resources/resource_compute_instance.go

```diff
@@ -938,14 +938,34 @@ func resourceComputeInstanceUpdate(d *schema.ResourceData, meta interface{}) err
 			return err
 		}
 
-		op, err := config.clientCompute.Instances.SetMetadata(project, zone, d.Id(), metadataV1).Do()
-		if err != nil {
-			return fmt.Errorf("Error updating metadata: %s", err)
-		}
+		// We're retrying for an error 412 where the metadata fingerprint is out of date
+		err = retry(
+			func() error {
+				// retrieve up-to-date metadata from the API in case several updates hit simultaneously. instances
+				// sometimes but not always share metadata fingerprints.
+				instance, err := config.clientComputeBeta.Instances.Get(project, zone, d.Id()).Do()
+				if err != nil {
+					return handleNotFoundError(err, d, fmt.Sprintf("Instance %s", d.Get("name").(string)))
```

Comment:
Should we be suppressing not found during update?

Reply:
This doesn't suppress it, it removes the resource from state if it isn't found. I guess it's technically the job of Read, but if we run into it here the resource is gone regardless and will be removed on the next refresh. What do you think should we defer to Read on this or handle it here too in case the instance falls out from under us?

Reply:
What I mean by suppress is that handleNotFoundError would d.SetId("") but still return a nil so we'd call d.SetPartial("metadata"). I guess it doesn't necessarily matter since it was removed from state and thus would not mark as having updated anything, but as far as I can tell, if the other operations failed due to 404, they would immediately error out without handling the 404.

Reply:
Alright, immediately returning the error like the others.